fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -90,8 +90,10 @@ def fit(self, X, y, **kwargs):
             logger.debug("Regression problem detected")
             is_classification_reduction = False
 
-        _, y_train, sensitive_features = _validate_and_reformat_input(
-            X, y, enforce_binary_labels=is_classification_reduction, **kwargs)
+        _, y_train, sensitive_features, control_features = \
```

Comment:
We should move the call to `_validate_and_reformat_input` into `Moment.load_data`. Exponentiated gradient is just a reduction, so it should be touching the data only minimally.

Reply:
There are.... issues with that. For example, `ExponentiatedGradient.fit()` uses `y_train.shape[0]` in its error estimates. That's relying on the data conversion being done in `_validate_and_reformat_input()`. I'm squinting at the usage of `y` inside `Lagrangian`, and I think that will be OK if it's with the unprocessed `y` .... but it's hard to be sure without a bunch of testing. There's a similar consideration in GridSearch.

I'm not saying that this isn't a good idea - and `_validate_and_reformat_input()` is turning into a bit of a monster too - but this is non-trivial.

Reply:
I think I have a way forward, but the surgery is going to be moderately major.

Reply:
You probably want something like `sklearn.utils.validation._num_samples`